src/FilePOMDPs.jl

```diff
@@ -0,0 +1,145 @@
+##### 
+struct FilePOMDP{Int64} <: POMDP{Int64, Int64, Int64} 
+    number_of_states::Int64
+    number_of_actions::Int64
+    number_of_observations::Int64
+
+    support_of_distribution::Set{Int64}
+    value_of_distribution::Vector{Float64}
+
+    discount::Float64
+
+    T::TransitionProb
+    O::ObservationProb
+    R::RewardLookUp
+end
+
+FilePOMDP(s::Int64, a::Int64, o::Int64, initial_state::InitialStateParam, discount::Float64, T::TransitionProb, O::ObservationProb, R::RewardLookUp)= FilePOMDP(s, a, o, support(initial_state), prob(initial_state), discount, T, O, R)  
+
+FilePOMDP(filename::String) = read_pomdp(filename, :FilePOMDP)
+
+states(m::FilePOMDP{Int64}) = 1:m.number_of_states
+stateindex(m::FilePOMDP{Int64}, i::Int64) = (i <= m.number_of_states) ? i : error("Querying states outside the allowable range.")
+
+actions(m::FilePOMDP{Int64}) = 1:m.number_of_actions
+actionindex(m::FilePOMDP{Int64}, i::Int64) = (i <= m.number_of_actions) ? i : error("Querying input outside the allowable range.")
+
+observations(m::FilePOMDP{Int64}) = 1:m.number_of_observations
+obsindex(m::FilePOMDP{Int64}, i::Int64) = (i <= m.number_of_observations) ? i : error("Querying observations outside the allowable range.")
+
+function initialstate(m::FilePOMDP{Int64})
+
+    if !isempty(m.value_of_distribution)
+        return SparseCat(states(m), m.value_of_distribution)
+    else
+        # @warn "No available initial condition."
+        return SparseCat(states(m), 1/m.number_of_states*ones(m.number_of_states))
+    end
+end
+
+transition(m::FilePOMDP{Int64}, s::Int64, a::Int64, sp::Int64) = m.T[(s,a,sp)]
```

Comment:
We should not define methods of `POMDPs.transition` that have different meaning from the documentation. i.e. all methods should return a distribution